### PR TITLE
🌱 Add more templating func to prowjob-gen

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
Required for CAPV templates improvements

### CODE REVIEW

File: hack/tools/prowjob-gen/generator.go [hunk 1]
```diff
@@ -195,3 +199,39 @@ func last(list any) any {
...
+		for i := 0; i < l; i++ {
+			item = l2.Index(i).Interface()
+			if reflect.DeepEqual(needle, item) {
+				return true, nil
+			}
+		}
+
+		return false, nil
+	default:
+		return false, fmt.Errorf("Cannot find has on type %s", tp)
```
Comment (chrischdi) [inline hack/tools/prowjob-gen/generator.go:235]:
```suggestion
		return false, fmt.Errorf("cannot find has on type %s", tp)
```

**Review by chrischdi — COMMENTED** [review]
/lgtm

File: hack/tools/prowjob-gen/generator.go [hunk 2]
```diff
@@ -192,6 +196,42 @@ func last(list any) any {
...
+		for i := 0; i < l; i++ {
+			item = l2.Index(i).Interface()
+			if reflect.DeepEqual(needle, item) {
+				return true, nil
+			}
+		}
+
+		return false, nil
+	default:
+		return false, fmt.Errorf("cannot find has on type %s", tp)
```
Comment (sbueringer) [inline hack/tools/prowjob-gen/generator.go:235]: nit: the error sounds a bit strange

Comment (fabriziopandini) [reply hack/tools/prowjob-gen/generator.go:235]: it is a copy paste from sprig, not sure we should/want to start forking...

Comment (sbueringer) [reply hack/tools/prowjob-gen/generator.go:235]: I'm fine to keep it as is. But the error message doesn't seem to make any sense :D

**Comment by sbueringer** [comment]
/lgtm
/approve

/hold

up toyou

File: hack/tools/prowjob-gen/generator.go (hunk 2 above)
Comment (sbueringer) [reply hack/tools/prowjob-gen/generator.go:235]: `cannot evaluate has on type %s (only arrays and slices are supported)` would make a lot more sense

**Pull request merged**